From 93c3f1b5a2efd5871e53705daedbb67642150c44 Mon Sep 17 00:00:00 2001
From: Julien Fontanet <julien.fontanet@isonoe.net>
Date: Wed, 2 Jun 2021 17:10:22 +0200
Subject: [PATCH 1/4] fix(XOA deploy): dont fail on missing XenStore entries

May fix https://twitter.com/cqchu/status/1396769270537785344

Entries may have already been removed by the appliance itself.

Signed-off-by: Julien Fontanet <julien.fontanet@isonoe.net>
---
 src/xenserver/opt/xensource/www/asset/deploy.js | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/xenserver/opt/xensource/www/asset/deploy.js b/src/xenserver/opt/xensource/www/asset/deploy.js
index 754cf40..ad9faa7 100644
--- a/src/xenserver/opt/xensource/www/asset/deploy.js
+++ b/src/xenserver/opt/xensource/www/asset/deploy.js
@@ -291,7 +291,9 @@ function deploy() {
           'xen-servers',
           'xoa-updater-credentials'
         ].map(key =>
-          call('VM.remove_from_xenstore_data', vmRef, `vm-data/${key}`)
+          call('VM.remove_from_xenstore_data', vmRef, `vm-data/${key}`).catch(error => {
+            console.warn('VM.remove_from_xenstore_data', key, error)
+          })
         )
       )
     })
-- 
2.30.2

